contrib/jms/src/test/java/com/google/cloud/dataflow/contrib/jms/JmsIOTest.java

```diff
@@ -0,0 +1,140 @@
+import com.google.cloud.dataflow.sdk.testing.DataflowAssert;
+import com.google.cloud.dataflow.sdk.testing.TestPipeline;
+import com.google.cloud.dataflow.sdk.transforms.Create;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+
+import org.apache.activemq.ActiveMQConnectionFactory;
+import org.apache.activemq.broker.BrokerService;
+import org.apache.activemq.store.memory.MemoryPersistenceAdapter;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+
+import javax.jms.Connection;
+import javax.jms.ConnectionFactory;
+import javax.jms.MessageConsumer;
+import javax.jms.MessageProducer;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+
+/**
+ * Test of the JmsIO.
+ */
+public class JmsIOTest {
+
+    private static final String BROKER_URL = "vm://localhost";
+
+    private BrokerService broker;
+    private ConnectionFactory connectionFactory;
+
+    @Before
+    public void startBroker() throws Exception {
+        System.out.println("Starting ActiveMQ broker on " + BROKER_URL);
+        broker = new BrokerService();
+        broker.setUseJmx(false);
+        broker.setPersistenceAdapter(new MemoryPersistenceAdapter());
+        broker.addConnector(BROKER_URL);
+        broker.setBrokerName("localhost");
+        broker.start();
+
+        // create JMS connection factory
+        System.out.println("Create JMS connection factory on " + BROKER_URL);
+        connectionFactory = new ActiveMQConnectionFactory(BROKER_URL);
+    }
+
+    @Test
+    public void testReadSingleMessage() throws Exception {
+
+        // produce message
+        System.out.println("Producing a test message on the broker ...");
+        Connection connection = connectionFactory.createConnection();
+        Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+        MessageProducer producer = session.createProducer(session.createQueue("test"));
+        TextMessage message = session.createTextMessage("This Is A Test");
+        producer.send(message);
+        producer.send(message);
+        producer.send(message);
+        producer.send(message);
+        producer.send(message);
+        producer.send(message);
+        producer.close();
+        session.close();
+        connection.close();
+
+        System.out.println("Creating test pipeline");
+        Pipeline pipeline = TestPipeline.create();
+
+        // read from the queue
+        System.out.println("Reading from the JMS queue and create PCollection");
+        PCollection<String> output =  pipeline.apply(
+                JmsIO.Read
+                        .connectionFactory(connectionFactory)
+                        .queue("test")
+                        .maxNumMessages(5));
+
+        System.out.println("Starting the pipeline");
+        pipeline.run();
+
+        System.out.println("Check assertion");
+        DataflowAssert.that(output).containsInAnyOrder(new String[]{ "test "});
```

Comment:
Hmm. I would have expected a test more like

```java
List<String> testMessages = ImmutableList.of("msg1", "msg2", "msg3");
...
for (String m : testMessages) {
    producer.send(session.createTextMessage(m));
}
...

...
Pipeline pipeline = ...;
PCollection<String> output = pipeline.apply(...);
DataflowAssert.that(output).containsInAnyOrder(testMessages);
pipeline.run();
```

Note that the `DataflowAssert.that()...` line must go **before** the pipeline is run. Otherwise it's not actually in the pipelien that is executed!

Note that this test should fail -- `test` is not the test message that you sent! -- but the assert is not being run.